public/js/pimcore/elementservice.js

```diff
@@ -83,9 +83,25 @@ pimcore.elementservice.deleteElementsComplete = function(options, response) {
 pimcore.elementservice.deleteElementCheckDependencyComplete = function (window, res, options) {
 
     try {
-        let message = res.batchDelete ? t('delete_message_batch') : t('delete_message');
+        let message = '';
+        if (res.batchDelete) {
+            message += sprintf(t('delete_message_batch'), res.itemResults.length) + "<br /><div>";
+            res.itemResults.forEach(function (item) {
+                if (res.itemResults.length > 0) {
+                    message += "<ul>";
+                    res.itemResults.forEach(function (item) {
+                        message += '<li>' + htmlspecialchars(item.path) + '<b>' + htmlspecialchars(item.key) + '</b></li>';
+                    })
+                    message += "</ul>";
+                }
+            })
```

Comment:
the suggestion previously applied wasn't right, the itemResults would be iterated esponentially

```suggestion
            if (res.itemResults.length > 0) {
                message += "<ul>";
                res.itemResults.forEach(function (item) {
                    message += '<li>' + htmlspecialchars(item.path) + '<b>' + htmlspecialchars(item.key) + '</b></li>';
                })
                message += "</ul>";
            }
```